Products/RhaptosPrint/RhaptosPrintTool.py

```diff
@@ -346,11 +359,32 @@ def getAlternateStyles(self):
                  {'title':'CCAP Business Ethics', 'id':'ccap-business-ethics'},
                  {'title':'CCAP SRM Business', 'id':'ccap-srm-business'}  
                ]
+
+        # Get list of css files available for Docbook/PrinceXML pipeline
+        all_css =[css[:-4] for css in
+                  os.listdir(self.getEpubDir()+'/css')
+                  if css.endswith('.css')]
+
+        # Remove any from the manual style list that do not have css behind it
+        for style_id in list([s['id'] for s in styles]):
+            if style_id not in all_css:
+                styles.remove(_style_by_id(style_id, styles))
+
         # Now fetch baking styles from the db, vi RhaptosModuleStorage
-        extra_styles = moduledb_tool.sqlGetPrintStyles()
-        if extra_styles:
-            import simplejson as json
-            styles.extend([json.loads(s[0]) for s in extra_styles])
+        recipes = [json.loads(s[0]) for s in moduledb_tool.sqlGetPrintStyles()]
+
+        # Only keep recipes that have an associated css
+        for recipe_id in list([r['id'] for r in recipes]):
+            if recipe_id not in all_css:
+                recipes.remove(_style_by_id(recipe_id, recipes))
+
+        # Delete from manual list, if w/ recipe also exists
```

Comment:
avoids the duplication of "Biology" vs "Biology (with recipe)" where both actually use the recipe, since they have the same print-style-id.